### PR TITLE
Fix uncatchable exception in sulu_categories twig extension

### LINKED ISSUE #4270: Twig sulu_categories throws exception if category with parent key does not exist
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | dev-develop (59812040b20170d5000b79a734d84670d3994202)
| Browser Version | Chrome 70

#### Actual Behavior

When using sulu_categories with a parent key that does not exist then the thrown exception is passed on to the frontend. Normally the searched for parentKey does exist but if it does not, it shouldn't break the whole frontend.

#### Expected Behavior

If the parent key is not found then it should just return an empty array.

#### Steps to Reproduce

{% sulu_categories(app.request.locale, 'missing-category-key') 

#### Possible Solutions

Catch the exception CategoryKeyNotFoundException and return an empty array instead.

### PR DESCRIPTION
**Pull request by AndreasA**
Fixes: #4270

| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #4270
| License | MIT

#### What's in this PR?

Fixed the mentioned ticket.

#### Why?

Fixed the mentioned ticket.

### CODE REVIEW

File: src/Sulu/Bundle/CategoryBundle/Twig/CategoryTwigExtension.php [hunk 1]
```diff
@@ -83,12 +93,21 @@ public function getCategoriesFunction($locale, $parentKey = null)
...
-                return $this->serializer->serialize($categories, $context);
+                try {
+                    $entities = $this->categoryManager->findChildrenByParentKey($parentKey);
+                    $categories = $this->categoryManager->getApiObjects($entities, $locale);
+                    $context = SerializationContext::create();
+                    $context->setSerializeNull(true);
+
+                    return $this->serializer->serialize($categories, $context);
+                } catch (CategoryKeyNotFoundException $exception) {
+                    $this->logger->error(
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/CategoryBundle/Twig/CategoryTwigExtension.php]: We are not logging errors in other cases when something don't exist when loaded over twig extension. So I would not do it here for consistents.

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
Thank you for the pull request can you rebase this and target the 2.2 branch?

File: src/Sulu/Bundle/CategoryBundle/Twig/CategoryTwigExtension.php (hunk 1 above)
Comment (AndreasA) [reply src/Sulu/Bundle/CategoryBundle/Twig/CategoryTwigExtension.php]: OK.

**Review by alexander-schranz — APPROVED** [review]

**Comment by AndreasA** [comment]
> Thank you for the pull request can you rebase this and target the 2.2 branch?

@alexander-schranz Done. and also removed the logger.

**Comment by AndreasA** [comment]
@alexander-schranz Just to let you know, I forgot to remove the unnecessary includes in the test case (LoggerInterface was still in it). I have removed it now as well.

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@AndreasA Thank you for fixing this!

**Pull request merged**